### PR TITLE
Change ordering of multipass v0 for backwards compatability

### PR DESCRIPTION
**Pull request by fuchsnj**
This is a temporary change to maintain backwards compatibility. Instead of getting all excluded matches and then finding all the exclusions, it is checked as strings are scanned, so the behavior of multipass v0 depends on the order strings are scanned.

Implementations of `Event` decide the order, so fields that tend to be excluded should normally be scanned first.

### CODE REVIEW

**Review by artslidd — APPROVED** [review]
Waiting for Vincent's review, but LGTM

File: sds/src/scanner/mod.rs
Comment (vinckama) [inline sds/src/scanner/mod.rs:1056]: Minor: let's validate that the path of the match is `a-match` so it's more explicit that `z-match` is the one excluded.

File: sds/src/scanner/mod.rs
Comment (vinckama) [inline sds/src/scanner/mod.rs:1043]: I would keep the exact same input for `a-match` and `z-match` so the only difference is the path
```suggestion
                SimpleEvent::String("abcdef".to_string()),
```

**Review by vinckama — APPROVED** [review]
LGTM with two minor comments

File: sds/src/scanner/mod.rs
Comment (fuchsnj) [reply sds/src/scanner/mod.rs:1043]: Fixed in a9d44ba1848f996b99c3d8a4bb83e59d8af51ba6

File: sds/src/scanner/mod.rs
Comment (fuchsnj) [reply sds/src/scanner/mod.rs:1056]: Fixed in a9d44ba1848f996b99c3d8a4bb83e59d8af51ba6

**Pull request merged**
